fdbserver/workloads/IncrementalBackup.actor.cpp

```diff
@@ -51,19 +54,50 @@ struct IncrementalBackupWorkload : TestWorkload {
 		return _start(cx, this);
 	}
 
-	virtual Future<bool> check(Database const& cx) { return true; }
+	virtual Future<bool> check(Database const& cx) {
+		if (clientId || !waitVersion) {
+			return true;
+		}
+		return _check(cx, this);
+	}
+
+	ACTOR static Future<bool> _check(Database cx, IncrementalBackupWorkload* self) {
+		state Reference<IBackupContainer> backupContainer;
+		state UID backupUID;
+		int waitResult =
+		    wait(self->backupAgent.waitBackup(cx, self->tag.toString(), false, &backupContainer, &backupUID));
+		TraceEvent("IBackupCheckWaitResult").detail("Result", waitResult);
+		state Reference<ReadYourWritesTransaction> tr(new ReadYourWritesTransaction(cx));
+		state Version v = wait(tr->getReadVersion());
+		loop {
+			BackupDescription desc = wait(backupContainer->describeBackup(true));
+			TraceEvent("IBackupVersionGate")
+			    .detail("MaxLogEndVersion", desc.maxLogEnd.present() ? desc.maxLogEnd.get() : invalidVersion)
+			    .detail("ContiguousLogEndVersion",
+			            desc.contiguousLogEnd.present() ? desc.contiguousLogEnd.get() : invalidVersion)
+			    .detail("TargetVersion", v);
+			if (!desc.contiguousLogEnd.present()) continue;
+			if (desc.contiguousLogEnd.get() >= v) break;
+			// Avoid spamming requests with a delay
+			wait(delay(5.0));
+		}
+		return true;
+	}
 
 	ACTOR static Future<Void> _start(Database cx, IncrementalBackupWorkload* self) {
-		// Add a commit both before the submit and restore to test that incremental backup
-		// can be performed on non-empty database
 		if (self->submitOnly) {
 			Standalone<VectorRef<KeyRangeRef>> backupRanges;
 			backupRanges.push_back_deep(backupRanges.arena(), normalKeys);
 			TraceEvent("IBackupSubmitAttempt");
 			try {
 				wait(self->backupAgent.submitBackup(cx, self->backupDir, 1e8, self->tag.toString(), backupRanges, false,
 				                                    false, true));
+				// Wait for backup container to be created and avoid race condition
+				wait(delay(60.0));
```

Comment:
This delay should not be necessary anymore, right?

Reply:
This delay is for a different purpose: the backup can be submitted but the cluster needs time to create the container. This is to avoid `describeBackup` from throwing an error saying that no backup exists

Reply:
Instead of sleeping, which relies on fragile timeouts, we could have a loop that keeps calling `waitBackup` and retrying until the container is created.

Reply:
I actually attempted to implement it that way at first. The reason that doesn't work is because we are using a differential backup (`stopWhenDone = false`), meaning our exit condition comes back early compared to a regular backup, giving false positives when the container isn't actually ready. In order for this idea to work, it would require a more involved change that could have other side effects. Jingyu and I discussed that this delay would be sufficient to avoid this race instead of making the test needlessly complex.